### PR TITLE
Convert CacheEpic to Promises

### PR DESCRIPTION
**Pull request by mmcgahan**
This is the next epic in the conversion sequence, one more PR to come

### CODE REVIEW

File: packages/mwp-api-state/src/cache/index.test.js [hunk 1]
```diff
@@ -25,31 +21,27 @@ function makeCacheEpic() {
...
-	const apiSuccessAction$ = ActionsObservable.of(MOCK_SUCCESS_ACTION);
-	return CacheEpic(apiSuccessAction$).toPromise().then(() => CacheEpic);
+	return CacheEpic(MOCK_SUCCESS_ACTION).then(() => CacheEpic);
 }
 
 function clearCacheEpic(CacheEpic) {
 	// clear the cache with CACHE_CLEAR
-	const clearAction$ = ActionsObservable.of({ type: 'CACHE_CLEAR' });
-	return CacheEpic(clearAction$).toPromise().then(() => CacheEpic);
+	return CacheEpic({ type: 'CACHE_CLEAR' }).then(() => CacheEpic);
```
Comment (eilinora) [inline packages/mwp-api-state/src/cache/index.test.js]: isn't this a const?

File: packages/mwp-api-state/src/sync/index.js [hunk 2]
```diff
@@ -64,85 +52,74 @@ export function getDeprecatedSuccessPayload(successes, errors) {
...
+		if (![LOCATION_CHANGE, SERVER_RENDER].some(type => type === action.type)) {
+			return Promise.resolve([]);
+		}
+		const { payload: location } = action;
+		const state = store.getState();
+		const { referrer = {} } = state.routing;
+		// inject request metadata from context, including `store.getState()`
+		const requestMetadata = {
+			referrer: referrer.pathname || state.config.entryPath || '',
+			logout: location.pathname.endsWith('logout'), // assume logout route ends with logout
```
Comment (eilinora) [inline packages/mwp-api-state/src/sync/index.js]: would be nice to have more notes here about why/what this is for or a link to those notes

File: packages/mwp-api-state/src/cache/index.test.js (hunk 1 above)
Comment (mmcgahan) [reply packages/mwp-api-state/src/cache/index.test.js]: so it would appear!

File: packages/mwp-api-state/src/sync/index.js (hunk 2 above)
Comment (mmcgahan) [reply packages/mwp-api-state/src/sync/index.js]: okay - this is actually sort of legacy code when we were thinking about how the app should handle logout, but it hasn't been implemented in any application so I'll make that clear

**Pull request merged**